Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -1412,11 +1412,30 @@ NonnullRefPtr<Expression> Parser::parse_secondary_expression(NonnullRefPtr<Expre
         return parse_call_expression(move(lhs));
     case TokenType::Equals:
         return parse_assignment_expression(AssignmentOp::Assignment, move(lhs), min_precedence, associativity);
-    case TokenType::Period:
+    case TokenType::Period: {
         consume();
-        if (!match_identifier_name())
+
+        String identifier {};
+        if (!match_identifier_name()) {
             expected("IdentifierName");
-        return create_ast_node<MemberExpression>({ m_state.current_token.filename(), rule_start.position(), position() }, move(lhs), create_ast_node<Identifier>({ m_state.current_token.filename(), rule_start.position(), position() }, consume().value()));
+        } else {
+            auto token = consume();
```

Comment:
Is this something that every identifier consuming place should do?
In that case we probably want to put it in some method since with this escaped identifiers will not work in other places.

Reply:
Yeah actually, I have that change made locally already - I will be storing the resolved identifier value in `Token`, which will return that value in `Token::value`, so all call sets get the right string.